### PR TITLE
[MNG-7095] Support for parallel builds when resuming

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

**Comment by gnodet** [comment]
I'll add another test

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java [hunk 1]
```diff
@@ -175,6 +179,112 @@ public void testShouldNotCrashWithoutProject() throws Exception
...
+            verifier.assertArtifactPresent( "org.apache.maven.its.mng5760", "module-b", "1.0", "jar" );
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-c", "1.0", "jar" );
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-d", "1.0", "jar" );
+        }
+        finally
+        {
+            verifier.resetStreams();
+        }
+
+        // Let module-b and module-c fail, if they would have been built...
```
Comment (MartinKanters) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java:214]: You are only letting module-b fail. I think the comment is not correct, right? (if so, please also correct the one below)

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java [hunk 2]
```diff
@@ -175,6 +179,112 @@ public void testShouldNotCrashWithoutProject() throws Exception
...
+
+        verifier.executeGoal( "org.apache.maven.plugins:maven-clean-plugin:clean" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-a", "1.0", "jar" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-b", "1.0", "jar" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-c", "1.0", "jar" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-d", "1.0", "jar" );
+
+        try
+        {
+            verifier.executeGoals( Arrays.asList( "org.apache.maven.plugins:maven-clean-plugin:clean", "install" ) );
```
Comment (MartinKanters) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java:250]: Clean is not necessary here as you have cleaned it in line 242

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java [hunk 3]
```diff
@@ -175,6 +179,112 @@ public void testShouldNotCrashWithoutProject() throws Exception
...
+            verifier.assertArtifactPresent( "org.apache.maven.its.mng5760", "module-b", "1.0", "jar" );
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-c", "1.0", "jar" );
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-d", "1.0", "jar" );
+        }
+        finally
+        {
+            verifier.resetStreams();
+        }
+
+        // Let module-b and module-c fail, if they would have been built...
```
Comment (MartinKanters) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java:266]: This should be module-a and module-b, right?

**Review by MartinKanters — CHANGES_REQUESTED** [review]
Nice ITs. Great that you added some comments to make the intention more clear.
Added some small review points, nothing major.

**Comment by MartinKanters** [comment]
I've rerun the GitHub Actions build on your maven PR, this should take these new ITs into account as well (as both branches are named the same):
https://github.com/apache/maven/actions/runs/561224481

**Comment by MartinKanters** [comment]
Also I don't mind it too much, but perhaps we could avoid `install` now, if MNG-4660 indeed has fixed the issue you described in MNG-7099.

**Comment by MartinKanters** [comment]
> I've rerun the GitHub Actions build on your maven PR, this should take these new ITs into account as well (as both branches are named the same):
> https://github.com/apache/maven/actions/runs/561224481

FYI, the ITs failed because your maven-integration-testing branch is not even with apache:master.

**Comment by gnodet** [comment]
I've fixed the comments and change the tests to rely on the `verify` goal rather than `install`.

**Comment by MartinKanters** [comment]
> I've fixed the comments and change the tests to rely on the `verify` goal rather than `install`.

Great. They are still running atm, but the results look good: https://github.com/apache/maven/runs/1890081377?check_suite_focus=true

**Comment by gnodet** [comment]
> > I've fixed the comments and change the tests to rely on the `verify` goal rather than `install`.
> 
> Great. They are still running atm, but the results look good: https://github.com/apache/maven/runs/1890081377?check_suite_focus=true

Is the CI build set up to run the tests against a branch of maven with the same name ?

**Comment by MartinKanters** [comment]
> > > I've fixed the comments and change the tests to rely on the `verify` goal rather than `install`.
> > 
> > 
> > Great. They are still running atm, but the results look good: https://github.com/apache/maven/runs/1890081377?check_suite_focus=true
> 
> Is the CI build set up to run the tests against a branch of maven with the same name ?

Yep, it checks whether the contributor who provides the PR has a matching branch on his/hers maven-integration-testing fork.
You can see it in the "Collect environment context variables" step in one of the integration-test tasks:
> Found a branch "MNG-7095" in fork "gnodet/maven-integration-testing", configuring this for the integration tests to be run against.

**Review by mthmulders — APPROVED** [review]
Nice work!

**Pull request closed without merging**